File: week5/youngjin/programmers_practice_test_42840.py
def solution(answers):
    # 학생 별 정답 패턴
    no1 = [1, 2, 3, 4, 5]
    no2 = [2, 1, 2, 3, 2, 4, 2, 5]
    no3 = [3, 3, 1, 1, 2, 2, 4, 4, 5, 5]

    # 학생 별 맞은 문제 수
    cnt = [0,0,0]
    
    for i in range(len(answers)):
        # 인덱스 순환하면서 검사
        if answers[i] == no1[i % len(no1)]: 
            cnt[0] +=1
        if answers[i] == no2[i % len(no2)]:
            cnt[1] +=1
        if answers[i] == no3[i % len(no3)]:
            cnt[2] +=1
            
    answer = [i+1 for i, value in enumerate(cnt) if value == max(cnt)] # 학생 번호는 1부터 시작, 인덱스(i) + 1
    
    return answer

File: week5/hanhee/programmers_practice_test_42840.cpp
#include <string>
#include <vector>
#include <algorithm> 
#include <iostream>

using namespace std;

vector<int> solution(vector<int> answers) {
    vector<int> answer;
    int n1[5]={1,2,3,4,5},
    n2[8]={2,1,2,3,2,4,2,5},
    n3[10]={3,3,1,1,2,2,4,4,5,5};
    int sum[3]={0,};
    
    for(int i=0; i<answers.size(); i++){
        int ans = answers[i];
        if(ans==n1[i%5]) sum[0]++;
        if(ans==n2[i%8]) sum[1]++;
        if(ans==n3[i%10]) sum[2]++;

    }
    
    int max = *max_element(sum, sum+3);
    
    for(int i=0; i<3; i++) {
        if(sum[i]==max) answer.push_back(i+1);
    }
    
    return answer;
}